derive-new-test/Cargo.lock

```diff
@@ -1,38 +0,0 @@
-[root]
```

Comment:
I update the Cargo.lock whenever I deliberately to a cargo update, which might be more or less often depending on the project.

I think the official advice is out of date, and IMO wrong. The reasoning is that developers get the same 'view' of the project as those who link the project as an upstream crate (because a checked-in Cargo.lock is not used by downstream crates). However, I think this does not actually help very much and IMO it is better that all developers on a project get the same view of the project, even if users suffer (which they shouldn't if everyone is doing back-compat properly).